packages/tests/test/reya_common/trade/PassivePool.fork.c.sol

```diff
@@ -255,4 +266,288 @@ contract PassivePoolForkCheck is BaseReyaForkTest {
         // withdraw 100 sUSDe from account
         executePeripheryWithdrawMA(user, userPk, 1, accountId, sec.susde, 100e18, sec.mainChainId);
     }
+
+    function check_PassivePoolWithDeusd() public {
+        mockFreshPrices();
+
+        (CollateralConfig memory collateralConfig, ParentCollateralConfig memory parentCollateralConfig,) =
+            ICoreProxy(sec.core).getCollateralConfig(1, sec.deusd);
+
+        vm.prank(sec.multisig);
+        collateralConfig.cap = type(uint256).max;
+        ICoreProxy(sec.core).setCollateralConfig(1, sec.deusd, collateralConfig, parentCollateralConfig);
+
+        (address user, uint256 userPk) = makeAddrAndKey("user");
+
+        uint256 sharePrice0 = IPassivePoolProxy(sec.pool).getSharePrice(sec.passivePoolId);
+
+        // add 3000 deUSD to the passive pool directly
+        deal(sec.deusd, address(sec.periphery), 3000e18);
+        mockBridgedAmount(dec.socketExecutionHelper[sec.deusd], 3000e18);
+        vm.prank(dec.socketExecutionHelper[sec.deusd]);
+        IPeripheryProxy(sec.periphery).depositExistingMA(
+            DepositExistingMAInputs({ accountId: sec.passivePoolAccountId, token: address(sec.deusd) })
+        );
+
+        // check that the new 3000 deUSD does not influence the share price
+        uint256 sharePrice1 = IPassivePoolProxy(sec.pool).getSharePrice(sec.passivePoolId);
+        assertApproxEqRelDecimal(sharePrice1, sharePrice0, 0.005e18, 18);
+
+        // make sure that the passive pool deposit works
+        deal(sec.usdc, sec.periphery, 10e6);
+        vm.prank(dec.socketExecutionHelper[sec.usdc]);
+        vm.mockCall(
+            dec.socketExecutionHelper[sec.usdc],
+            abi.encodeCall(ISocketExecutionHelper.bridgeAmount, ()),
+            abi.encode(10e6)
+        );
+
+        IPeripheryProxy(sec.periphery).depositPassivePool(
+            DepositPassivePoolInputs({ poolId: sec.passivePoolId, owner: user, minShares: 0 })
+        );
+
+        uint256 sharesIn = IPassivePoolProxy(sec.pool).getAccountBalance(sec.passivePoolId, user);
+
+        // make sure that the passive pool withdrawal works
+        vm.prank(user);
+        uint256 amountOut = IPassivePoolProxy(sec.pool).removeLiquidity(sec.passivePoolId, sharesIn, 0);
+        assertApproxEqAbsDecimal(amountOut, 10e6, 10, 6);
```

Comment:
consider a fuzz test with an array of inputs -> tokens and amounts
perform a series of deposit and withdraw passive pool liquidity operations and make sure the share price does not go down and only goes up a bit as a result of rounding

Reply:
fuzz test with an array of target ratios, we set the target ratios and in the function we check and make sure share price doesn't go down

Reply:
try and perform a withdrawal of more than you have in any token